### DISCUSSION TITLE
支持用lua编写TrafficMonitor插件

### DISCUSSION THREAD

**Discussion by compilelife** [zhongyang219/TrafficMonitor#1789] [Ideas]
c++写插件有门槛，我实现了个[TrafficMontorLuaPlugin](https://github.com/compilelife/TrafficMonitorLuaPlugin)，支持用lua写插件。

目前只支持文本渲染，自定义绘图这个lua支持起来有点繁琐，用的人多再考虑。不过我让lua返回的文本能用`[smile]`这样的标记来在文本中显示类emoji的图标

来看看两个例子吧：[调用nvidia-smi显示显卡使用率](https://github.com/compilelife/TrafficMonitorLuaPlugin/blob/master/plugins/lua/%E6%98%BE%E5%8D%A1%E7%9B%91%E6%B5%8B.lua), [实时显示今天挣了多少钱](https://github.com/compilelife/TrafficMonitorLuaPlugin/blob/master/plugins/lua/%E8%B5%9A%E9%92%B1.lua)

下面是”调用nvidia-smi显示显卡使用率“的完整代码：
```lua
name="nvidia 显卡监测"
sample="显卡: 100%"
interval=3

function onUpdate()
	return "显卡: "..tf.runCmdLine("nvidia-smi.exe --query-gpu=utilization.gpu --format=csv,noheader")
end

function onClick()
end
```

如果感兴趣的人多，我将继续优化TrafficMontorLuaPlugin

**Comment by zhongyang219** [comment]
Lua插件非常棒，我已经在TrafficMonitor插件下载页面添加了Lua插件的相关链接，希望能够被更多人使用
